### PR TITLE
Make VR patch work for any(ish) Outer Wilds version

### PR DESCRIPTION
**Pull request by Raicuparta**
Credit to @TAImatem and @misternebula

### CODE REVIEW

**Comment by amazingalek** [comment]
Bump version please :)

**Comment by Raicuparta** [comment]
@amazingalek this is still just a draft please don't look at my work of art while it's unfinished 🙈🙈🙈 I'll need two patches for this one, one for Steam and one for Epic.

**Comment by amazingalek** [comment]
image

**Comment by TAImatem** [comment]
Looks like all bugs are fixed now

File: OWML.Patcher/BinaryPatcher.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                else
+                {
+                    patchZoneMatch = 0;
+                }
+                if (patchZoneMatch == patchZoneBytes.Length)
+                {
+                    return i + PatchStartZoneOffset;
+                }
+            }
+            throw new Exception("Could not find patch zone in globalgamemanagers. This probably means the VR patch needs to be updated.");
```
Comment (amazingalek) [inline OWML.Patcher/BinaryPatcher.cs:87]: It's recommended to use an exception which inherits from Exception and says something about what's up. Is there one that makes sense to use?

Comment (Raicuparta) [reply OWML.Patcher/BinaryPatcher.cs:87]: I looked for it, but couldn't find anything that made sense. Open to suggestions. I used FileNotFoundException where it made sense.

File: OWML.Patcher/VrPatcher.cs [hunk 2]
```diff
@@ -1,129 +1,37 @@
...
         {
             _owmlConfig = owmlConfig;
             _writer = writer;
-            _sha = SHA256.Create();
         }
 
         public void PatchVR(bool enableVR)
         {
-            PatchGlobalManager(enableVR);
+            var binaryPatcher = new BinaryPatcher(_owmlConfig, _writer);
```
Comment (amazingalek) [inline OWML.Patcher/VrPatcher.cs:21]: I'd like this to be newed up either outside of this class and injected into the constructor, or newed up in the constructor :)

File: OWML.Patcher/BinaryPatcher.cs (hunk 1 above)
Comment (amazingalek) [reply OWML.Patcher/BinaryPatcher.cs:87]: even better, then you can make your own 🥳 
PatchNotFoundException? 😁

File: OWML.Patcher/VrPatcher.cs (hunk 2 above)
Comment (amazingalek) [reply OWML.Patcher/VrPatcher.cs]: 👍

**Comment by TAImatem** [comment]
wait, why does it allow merge after only one approval

**Pull request merged**